lib/saml_idp/request.rb

```diff
@@ -31,48 +31,129 @@ def initialize(raw_xml = "")
       self.raw_xml = raw_xml
     end
 
+    def logout_request?
+      logout_request.nil? ? false : true
+    end
+
+    def authn_request?
+      authn_request.nil? ? false : true
+    end
+
     def request_id
-      authn_request["ID"]
+      if authn_request?
+        authn_request["ID"]
+      elsif logout_request?
+        logout_request["ID"]
```

Comment:
Our app uses the boolean methods to determine the flavor of the request. It seems reasonable to me that there be some public way to determine that flavor, since this class is doing double-duty.

I did add a `request` method at your suggestion, in 6ac5ea86e24c2f96b5e60ef25323bce2a3d8161b